### PR TITLE
Add proxy, more session control

### PR DESCRIPTION
**Pull request by makarworld**
## Changes:
- add proxy support to all requests
- add cookies arg to init (if user saved session and want continue with them)
- add autologout arg to init (if user want to save cookies and use it again later)
- edit login method, if user provide username, password and steam guard when initialize class, method should not request
them
- edit login method, if session is alive, skip login method

## Example:
```python
from steampy.client import SteamClient

client = SteamClient(
    api_key='XXXXX5FXXDDXXXXXABECXXB0C86XXXXX', 
    username='steam_login', 
    password='steam_password', 
    steam_guard='account.maFile',
    proxies={
        "http": "http://login:password@host:port",
        "https": "http://login:password@host:port",
    }, 
    autologout = False)

client.login()
print(client.is_session_alive())
```

## Session Control:
Also i add option `autologout` in `__init__` function for more session control.
When `autologout` is True, session will be ended with program end
When `autologout` is False, session will be not end and you may re-use cookies later

### Now you can save session cookies and re-use it later.
```python
import json
cookies = client._session.cookies.get_dict()
with open('steam_cookies.json', 'w', encoding='utf-8') as f:
    json.dump(f, cookies)
```

### re-use example:
```python
from steampy.client import SteamClient
import json

with open('steam_cookies.json', 'r', encoding='utf-8') as f:
    cookies = json.load(f)
    
client = SteamClient(
    api_key = ..., 
    username = ..., 
    password = ..., 
    steam_guard = ...,
    proxies = ...,
    cookies = cookies,
    autologout = False)
    
print(client.is_session_alive())
```

### CODE REVIEW

**Comment by makarworld** [comment]
## Changes:
- add proxy support to all requests
- add cookies arg to init (if user saved session and want continue with them)
- add autologout arg to init (if user want to save cookies and use it again later)
- edit login method, if user provide username, password and steam guard when initialize class, method should not request
them
- edit login method, if session is alive, skip login method

## Example:
```python
from steampy.client import SteamClient

client = SteamClient(
    api_key='XXXXX5FXXDDXXXXXABECXXB0C86XXXXX', 
    username='steam_login', 
    password='steam_password', 
    steam_guard='account.maFile',
    proxies={
        "http": "http://login:password@host:port",
        "https": "http://login:password@host:port",
    }, 
    autologout = False)

client.login()
print(client.is_session_alive())
```

## Session Control:
Also i add option `autologout` in `__init__` function for more session control.
When `autologout` is True, session will be ended with program end
When `autologout` is False, session will be not end and you may re-use cookies later

### Now you can save session cookies and re-use it later.
```python
import json
cookies = client._session.cookies.get_dict()
with open('steam_cookies.json', 'w', encoding='utf-8') as f:
    json.dump(f, cookies)
```

### re-use example:
```python
from steampy.client import SteamClient
import json

with open('steam_cookies.json', 'r', encoding='utf-8') as f:
    cookies = json.load(f)
    
client = SteamClient(
    api_key = ..., 
    username = ..., 
    password = ..., 
    steam_guard = ...,
    proxies = ...,
    cookies = cookies,
    autologout = False)
    
print(client.is_session_alive())
```

File: steampy/client.py [hunk 1]
```diff
@@ -29,21 +31,47 @@ def func_wrapper(self, *args, **kwargs):
...
+            if not isinstance(proxies, dict) :
+                raise TypeError('proxies must be a dict. Example: \{"http": "http://login:password@host:port"\, "https": "http://login:password@host:port"\}')
+            
+            # ping to steamcommunity.com
+            proxy_status = ping_proxy(proxies)
+
+            if proxy_status is True:
+                self._session.proxies.update(proxies)
+
+        self.steam_guard = guard.load_steam_guard(steam_guard)
```
Comment (bukson) [inline steampy/client.py:48]: I dont like it readed here, it is loaded in login function and thus SteamClient can be used for some cases with API key and without guard. Guard can be hard to obtain so this is braking change.

**Comment by bukson** [comment]
Please make 1 commit out of all those changes so in any case it is easily to revert it.

**Comment by makarworld** [comment]
> Please make 1 commit out of all those changes so in any case it is easily to revert it.

Okay

File: steampy/client.py (hunk 1 above)
Comment (makarworld) [reply steampy/client.py]: fixed

**Comment by bukson** [comment]
I have a small problem with rebasing it with 1 commit onto current master, I will try to deal with it next week and make new version 0.92 of steampy when it is ready.

**Comment by bukson** [comment]
Ok merged it manually with master, closing this.

Autologout variable was delete, if you dont want to log out, just dont use **with** statement.

**Pull request closed without merging**